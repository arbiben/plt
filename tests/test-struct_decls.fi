$tests passing structs into functions $
$demonstrates that structs can be declared globally or locally$
$as well as arrays$
$and that global structs can be assigned in functions$
$local structs can be passed into functions, manipulated, and returned with changes$


struct person{
    int age?
    int occupation?
    arr str 5 family?
    str secret?
}

struct person o?
arr str 4 bm?

int main(){
    struct person ori?
    struct person ggg?
    ori.age = 5?
    bm[0] = "hello!"?
    ggg = getOlder(ori)?
    print(ggg.age)?
    printstring(ggg.secret)?
    printstring(ggg.family@0)?
    printstring(bm@1)?

}


struct person getOlder(struct person om){
    o.age = 7?
    o.secret = "hi"?
    o.family[0] = "hi"?
    printstring(bm@0)?
    bm[1] = "/nyo!!!/n"?
    printstring(bm@1)?
    return o?
}

