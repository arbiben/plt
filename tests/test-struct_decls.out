hello!/nyo!!!/n7hihi/nyo!!!/n